### DISCUSSION TITLE
Support Valibot

### DISCUSSION THREAD

**Discussion by gustavopch** [edmundhung/conform#242] [Ideas]
Valibot ([repo](https://github.com/fabian-hiller/valibot), [announcement](https://www.builder.io/blog/introducing-valibot)) is a new validation library that seeks to provide a similar experience to Zod, but in a tree-shakeable way. From their annoucement:

> For example, to validate a simple login form, Zod requires [11.51 KB](https://bundlejs.com/?q=zod&treeshake=%5B%7B+object%2Cstring+%7D%5D) whereas Valibot requires only [0.7 KB](https://bundlejs.com/?q=valibot&treeshake=%5B%7B+email%2CminLength%2Cobject%2Cstring+%7D%5D).

Maybe consider adding support for it.

**Comment by edmundhung** [comment]
Hi @gustavopch,

I think it is still early to consider integrating with Valibot. As Conform is not only using the schema for validation, but also deriving constraints through an extensive schema introspection. It will require fair effort to integrate a new solution and maintain it.

But I am open to re-evaluate it in the future.

**Comment by gustavopch** [comment]
Fair enough, thanks for answering. Feel free to leave the issue open or close it if you prefer.

**Comment by chimame** [comment]
I created a simple sample that supports valibot based on `@conform-to/zod`.
I would be happy if you could give me some advice on making an adapter.

https://github.com/chimame/conform-to-valibot

**Reply by edmundhung** [reply]
Thanks for working on this! I thought it wasn't possible to introspect the valibot schema for type coercion but it looks like I was wrong. If you need any support, feel free to ping me on twitter or discord :)

**Comment by naourass** [comment]
Any update on using Conform with Valibot?

**Reply by vladinator1000** [reply]
You can try out https://github.com/chimame/conform-to-valibot

**Comment by chimame** [comment]
Please give us some more time to develop support for Valibot.
We are talking with edmundhung and considering including it as an official package.

**Comment by chimame** [comment]
Sorry for the long wait.
We've released the version that officially supports valibot.

https://www.npmjs.com/package/@conform-to/valibot
